design/status-design-update.md

```diff
@@ -0,0 +1,1166 @@
+# HTTPProxy status redesign
+
+## Background
+
+Given its complexity, Contour's HTTPProxy CRD currently has insufficient information about what its status surfaced to the user.
+
+Currently, it only supports a `currentStatus` and `description`, which are particularly poor user experience around inclusion and conditions.
+
+In particular, the following questions about HTTPProxy are particularly problematic at the moment:
+
+- What happens when *some* parts of a HTTPProxy configuration are invalid? What if one include is invalid, and the other is not? How can the user be aware of what's happening?
+- What happens if there are problems on inclusion, particularly condition conflicts?
+- What happens if there is a problem with composable HTTPProxy use, that is, including the same HTTPProxy object from multiple parents.
```

Comment:
Today if a parent is `invalid` then the children become `orphaned` since they essentially have no one delegated. I bring this up because it sounds like that logic might be changing.